src/connector/src/sink/iceberg.rs

```diff
@@ -0,0 +1,195 @@
+            .map_err(|e| SinkError::Config(anyhow!(e)))?;
+
+        if config.r#type != SINK_TYPE_APPEND_ONLY && config.r#type != SINK_TYPE_UPSERT {
+            return Err(SinkError::Config(anyhow!(
+                "`{}` must be {}, or {}",
+                SINK_TYPE_OPTION,
+                SINK_TYPE_APPEND_ONLY,
+                SINK_TYPE_UPSERT
+            )));
+        }
+        Ok(config)
+    }
+}
+
+pub struct IcebergSink {
+    config: IcebergConfig,
+    table: Table,
+    schema: Schema,
+}
+
+impl Debug for IcebergSink {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("IcebergSink")
+            .field("config", &self.config)
+            .finish()
+    }
+}
+
+impl IcebergSink {
+    fn parse_bucket_and_root_from_path(path: &str) -> Result<(String, String)> {
+        let prefix = if path.starts_with("s3://") {
+            "s3://"
+        } else if path.starts_with("s3a://") {
+            "s3a://"
+        } else if path.starts_with("s3n://") {
+            "s3n://"
+        } else {
+            return Err(SinkError::Config(anyhow!(
+                "Invalid path prefix: {}. Only support s3://, s3a://, s3n:// now.",
+                path
+            )));
+        };
+
+        let path = path.trim_start_matches(prefix);
+        let (bucket, root) = path.split_once('/').unwrap_or((path, ""));
+        Ok((bucket.to_string(), format!("/{}", root)))
+    }
+
+    pub async fn new(config: IcebergConfig, schema: Schema) -> Result<Self> {
+        let (bucket, root) = Self::parse_bucket_and_root_from_path(&config.path)?;
+
+        let mut builder = S3::default();
+
+        // Sink will not load config from file.
+        builder.disable_config_load();
+
+        builder
+            .root(&root)
+            .bucket(&bucket)
+            .endpoint(&config.endpoint)
+            .access_key_id(&config.access_key)
+            .secret_access_key(&config.secret_key);
+
+        if let Some(region) = &config.region {
+            builder.region(region);
+        }
+
+        let op = opendal::Operator::new(builder)
+            .map_err(|err| SinkError::Config(anyhow!("{}", err)))?
+            .finish();
+
+        let table = Table::open_with_op(op).await?;
+
+        Ok(Self {
+            config,
+            table,
+            schema,
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl Sink for IcebergSink {
+    /// TODO: Can't support commit coordinator now.
+    type Coordinator = DummySinkCommitCoordinator;
+    type Writer = TaskWriter;
+
+    async fn validate(&self, _connector_rpc_endpoint: Option<String>) -> Result<()> {
+        // We used icelake to write data into storage directly, don't need to validate.
+        Ok(())
+    }
+
+    async fn new_writer(&mut self, _writer_param: SinkWriterParam) -> Result<Self::Writer> {
+        // TODO: check consistency between schema and table schema.
+        self.table
+            .task_writer()
```

Comment:
Because in icelake, when we create a task writer, it will add the task_id
```
    /// Return a task writer used to write data into table.
    pub async fn task_writer(&mut self) -> Result<TaskWriter> {
        ...
        self.task_id += 1;
        Ok(task_writer)
    }
```

Reply:
Do we prepare make it be a shared-object. Otherwise, maybe RefCell🤔?